apis/compute/v1beta1/subnetwork_types.go

```diff
@@ -45,147 +43,72 @@ func (v *NetworkURIReferencerForSubnetwork) Assign(res resource.CanReference, va
 type SubnetworkParameters struct {
-	// Description: An optional description of this resource. Provide this
-	// property when you create the resource. This field can be set only at
-	// resource creation time.
-	// +optional
-	Description string `json:"description,omitempty"`
-
-	// EnableFlowLogs: Whether to enable flow logging for this subnetwork.
-	// If this field is not explicitly set, it will not appear in get
-	// listings. If not set the default behavior is to disable flow logging.
-	// +optional
-	EnableFlowLogs bool `json:"enableFlowLogs,omitempty"`
-
 	// IPCIDRRange: The range of internal addresses that are owned by this
 	// subnetwork. Provide this property when you create the subnetwork. For
 	// example, 10.0.0.0/8 or 192.168.0.0/16. Ranges must be unique and
 	// non-overlapping within a network. Only IPv4 is supported. This field
 	// can be set only at resource creation time.
 	IPCidrRange string `json:"ipCidrRange"`
 
-	// Name: The name of the resource, provided by the client when initially
-	// creating the resource. The name must be 1-63 characters long, and
-	// comply with RFC1035. Specifically, the name must be 1-63 characters
-	// long and match the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?`
-	// which means the first character must be a lowercase letter, and all
-	// following characters must be a dash, lowercase letter, or digit,
-	// except the last character, which cannot be a dash.
-	Name string `json:"name"`
-
 	// Network: The URL of the network to which this subnetwork belongs,
 	// provided by the client when initially creating the subnetwork. Only
 	// networks that are in the distributed mode can have subnetworks. This
 	// field can be set only at resource creation time.
-	Network string `json:"network,omitempty"`
+	// +immutable
+	Network *string `json:"network,omitempty"`
 
 	// NetworkRef references to a Network and retrieves its URI
 	NetworkRef *NetworkURIReferencerForSubnetwork `json:"networkRef,omitempty" resource:"attributereferencer"`
 
+	// Region: URL of the region where the Subnetwork resides. This field
+	// can be set only at resource creation time.
+	// +optional
+	Region string `json:"region"`
+
+	// Description: An optional description of this resource. Provide this
+	// property when you create the resource. This field can be set only at
+	// resource creation time.
+	// +optional
+	// +immutable
+	Description *string `json:"description,omitempty"`
+
+	// EnableFlowLogs: Whether to enable flow logging for this subnetwork.
+	// If this field is not explicitly set, it will not appear in get
+	// listings. If not set the default behavior is to disable flow logging.
+	// +optional
+	EnableFlowLogs *bool `json:"enableFlowLogs,omitempty"`
+
 	// PrivateIPGoogleAccess: Whether the VMs in this subnet can access
 	// Google services without assigned external IP addresses. This field
 	// can be both set at resource creation time and updated using
 	// setPrivateIPGoogleAccess.
```

Comment:
Confirmed patch call will fail if `privateIPGoogleAccess` is included

Reply:
> This field can be both set at resource creation time and updated using setPrivateIPGoogleAccess.

On a related note, I wonder how we should handle this kind of detail in our docstring comments. It seems like it could be misleading to Crossplane users, who may interpret "setPrivateIPGoogleAccess" as a Crossplane feature or functionality.

Reply:
I think we can just omit that sentence. For Crossplane users, that's the case for all non-immutable fields, it's just the implementation detail to call `setPrivateIPGoogleAccess`